packages/@sanity/codegen/src/typescript/moduleResolver.ts

```diff
@@ -0,0 +1,65 @@
+import {readFile, stat} from 'node:fs/promises'
+import path from 'node:path'
+
+// This is a custom implementation of require.resolve that takes into account the paths
+// configuration in tsconfig.json. This is necessary if we want to resolve paths that are
+// custom defined in the tsconfig.json file.
+// Resolving here is best effort and might not work in all cases.
+export async function getResolver(): Promise<NodeJS.RequireResolve> {
+  // find tsconfig.json
+  let resolveMap: Record<string, string[]> = {}
+  const cwd = process.cwd()
+  const tsconfigPath = path.join(cwd, 'tsconfig.json')
+  const tsconfigFile = await stat(tsconfigPath).catch((err) => {
+    if (err.code === 'ENOENT') {
+      return null
+    }
+    throw err
+  })
+
+  if (tsconfigFile) {
+    const tsconfigContent = await readFile(tsconfigPath)
+    try {
+      const tsconfigConfig = JSON.parse(tsconfigContent.toString())
+      if ('compilerOptions' in tsconfigConfig && 'paths' in tsconfigConfig.compilerOptions) {
+        resolveMap = tsconfigConfig.compilerOptions.paths as Record<string, string[]>
+      }
+    } catch (error) {
+      // ignore errors for now..
```

Comment:
some more details here would be great to add for the next reader. E.g. does "for now" mean that we should improve later? Why is it acceptable to silently swallow any error here?

Reply:
Good point! Added a debug log if this breaks. That way users can get some more context by turning on debug logging.